### ISSUE TITLE
Licensing question

### ISSUE THREAD

**Issue by ttlappalainen** [nox771/i2c_t3#25]
I have been using you library on my boat project. One of them may become commercial and then I will be in trouble with GPL. I changed my NMEA2000 library to MIT long time ago, when someone asked me to do that. How about your library, does it need to be GPL or could it be changed to MIT?

**Comment by nox771** [comment]
Yes this is something I plan on doing.  The current license is not actually GPL, it is LGPL.  This was a carryover from when the v1 code was originally created (it borrowed from existing Teensyduino Wire source at the time).  However since then it has been heavily rewritten, so I'm not even sure anything original is left.  Others have asked this also.  I was planning on swapping the header for MIT or BSD or something similar to eliminate concerns.  I just haven't had time to look into this yet.  As soon as I can find some time I'll investigate.

**Comment by ttlappalainen** [comment]
Thanks. Nice to here. Actually LGPL shouls be enough, since it does not require full source publishing. GPL is for expecting harms and feeding layers.

**Comment by mbenkmann** [comment]
I'm starting a new project and I'm hoping that I might be able to produce and sell a few units. I don't want to deal with the hassle of LGPL. The LGPL would require me to provide the buyer with everything he needs to rebuild the project. Since I'm using PlatformIO which is doing a lot of stuff behind the scenes I don't even know how to properly comply with the LGPL without providing a complete copy of my PlatformIO installation. And no matter what the answer to that question is, there's the issue of providing the buyer with the files. Do I send them via email? Then I would need some kind of return receipt or confirmation. I'm not going to set up and maintain a download website just to sell a few units of hardware. 
And I'm afraid if I do provide buyers with the files some might actually want to use them and ask me for support. While the LGPL does not require me to provide such support I can hardly tell customers to go F themselves.
So in summary, I'd rather invest the time to port https://github.com/piconomix/piconomix-fwlib to Teensy than to deal with LGPL compliance.

You say you plan to relicense your library. What exactly is keeping you? Is it just the hassle of replacing the headers in the source code and add the LICENSE? I can do that for you and submit a pull request.
This is kind of urgent for me since as I've said I'm starting a new project and it would not be a smart idea to start with i2c_t3 and then find myself in the situation that I need to switch over to fwlib later.

**Comment by FrankBoesing** [comment]
I've rarely read anything so bold.
What do you offer him for your demand?

**Comment by mbenkmann** [comment]
> I've rarely read anything so bold.
> What do you offer him for your demand?

No one is demanding anything here. What are you talking about? I am OFFERING my help in performing a licensing change he says he has planned. What's your problem?

**Comment by tonton81** [comment]
royalties :)

**Comment by nox771** [comment]
Well I had this penciled in for next July, is that a problem?

Actually the simple truth is that I've been very busy elsewhere (day job, having food on table, roof overhead, etc), and this just had not made it to the top of the priority list yet.  But in the near term this situation is only going to get worse, so I'll push this fix within the next day or so, in order to close this.

**Comment by mbenkmann** [comment]
As I've said, if you need any help with the boring mechanical parts of replacing the license, just tell me and I'll help. As for my project, I'll probably be finished planning end of December. If by that point i2c_t3's license has been changed, I'll use that. If not, I'll use a different i2c library (probably piconomix-fwlib).

**Comment by nox771** [comment]
I've uploaded a new version with MIT license.  Closing this.

**Comment by mbenkmann** [comment]
You should add a LICENSE file so that github picks up on the license.
